src/models/User.js

```diff
@@ -0,0 +1,60 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+
+const UserSchema = new mongoose.Schema({
+  firstName: {
+    type: String,
+    required: [true, "Please Provide Your First Name"],
+    maxlength: 50,
+    minlength: 3,
+  },
+  lastName: {
+    type: String,
+    required: [true, "Please Provide Your Last Name"],
+    maxlength: 50,
+    minlength: 3,
+  },
+  email: {
+    type: String,
+    required: [true, "Please Provide Email"],
+    match: [
+      /^([\w-\.]+@([\w-]+\.)+[\w-]{2,4})?$/,
+      "Please Provide a Valid Email",
+    ],
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: [true, "Please provide password"],
+    validate: {
+      validator: function (password) {
+        // Password must contain at least one lowercase letter, one uppercase letter, one number, and one special character
+        return /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[\W_]).{6,}$/.test(password);
+      },
+      message: "Password must contain at least one lowercase letter, one uppercase letter, one number, and one special character",
+    },
+  },
+  role: {
+    type: String,
+    enum: ["tutor", "parent"]
+  }
+});
+
+UserSchema.pre("save", async function (next) {
+  if (!this.isModified("password")) {
```

Comment:
as I correct understood this code used for changing password. It is strach goal.

Reply:
It is. I'll keep it. If we wouldn't have time for it, I'll remove it